### PR TITLE
[RTC-40] Add VAD chapter

### PR DESCRIPTION
**Pull request by daniel-jodlos**

### CODE REVIEW

File: src/webrtc/vad.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+All well and good, but how that translates into something readable? Well, let's start with "highest-intensity signal encodable by the payload format".
+It's simply the loudest volume that you could encode into the given codec, usually OPUS.
+
+Then there is an issue of decibels.
+**!!WARNING!! Physics intensifies**
+There are two things that you need to know about decibels:
+1. It's a unit of relative intensity, meaning that you need to define a base energy level.
+In the case of the commonly used dB scale that relative number is roughly the quietest sound that a human can hear.
+In the case of `dBox`, this level is the loudest sound that can be encoded.
```
Comment (mickel8) [inline src/webrtc/vad.md]: I feel like this is a duplication of the first two paragraphs

```
`dBov` is defined as the level, in decibels, relative to the overload point of the system.
In this context, an overload point is a highest-intensity signal encodable by the payload format.

All well and good, but how that translates into something readable? Well, let's start with "highest-intensity signal encodable by the payload format".
It's simply the loudest volume that you could encode into the given codec, usually OPUS.
```

File: src/webrtc/vad.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Active Speaker Detection
```
Comment (mickel8) [inline src/webrtc/vad.md:1]: Could we start a new sentence in a new line? It's easier to see changes in subsequent PRs. The other thing is enumeration, you can simply write:

```
1.
1.
1.
1.
```

File: src/webrtc/vad.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+### What is `dBov`
+`dBov` is defined as the level, in decibels, relative to the overload point of the system.
+In this context, an overload point is a highest-intensity signal encodable by the payload format.
+
+All well and good, but how that translates into something readable? Well, let's start with "highest-intensity signal encodable by the payload format".
+It's simply the loudest volume that you could encode into the given codec, usually OPUS.
+
+Then there is an issue of decibels.
+**!!WARNING!! Physics intensifies**
```
Comment (mickel8) [inline src/webrtc/vad.md]: This is not displayed in a separate line. Did you run mdbook :thinking: ?

File: src/webrtc/vad.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+There are two things that you need to know about decibels:
+1. It's a unit of relative intensity, meaning that you need to define a base energy level.
+In the case of the commonly used dB scale that relative number is roughly the quietest sound that a human can hear.
+In the case of `dBox`, this level is the loudest sound that can be encoded.
+2. Decibel is a logartmic value.
+The base of the logarithm differs on application, but usually, it's 10.
+Simply put, the `db` value tells you how many times you need to multiply the base level by 10 to get the measured value.
+In the case of negative numbers, we divide by 10 instead of multiplying.
```
Comment (mickel8) [inline src/webrtc/vad.md]: IMO, that's too basic

File: src/webrtc/vad.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+However, human speech doesn't have a constant volume.
+Humans also take small breaks between words.
+At 10 Hz, you can detect these and so, considering only one packet at a time isn't good enough, we need to look at the bigger picture.
+
+The simplest and most effective way to implement it is to use a time window.
+You then calculate the average volume of the packet and apply the threshold to that value.
+If the threshold is exceeded, speech is marked on the audio track.
+
+There are two parameters in this algorithm that you can tweak:
+1. `threshold` to tweak the sensitivity concerning the volume. You should tweak it if you find that your implementation doesn't detect actual speech at all, meaning that it is too high or, on the contrary, it interprets background noise as speech.
```
Comment (mickel8) [inline src/webrtc/vad.md]: Try to write shorter lines

File: src/webrtc/vad.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+2. Time window duration to tweak sensitivity to short, but loud sounds.
```
Comment (mickel8) [inline src/webrtc/vad.md]: Didn't we miss a wor here?
```suggestion
2. Time window duration to tweak sensitivity to detect short, but loud sounds.
```

**Review by mickel8 — CHANGES_REQUESTED** [review]

File: src/webrtc/vad.md (hunk 3 above)
Comment (daniel-jodlos) [reply src/webrtc/vad.md]: Yes, it wasn't supposed to be displayed in a separate line. Actually, we might consider removing this line :stuck_out_tongue:

File: src/webrtc/vad.md (hunk 2 above)
Comment (daniel-jodlos) [reply src/webrtc/vad.md:1]: > The other thing is enumeration, you can simply write:

I know, but I don't like this style of enumeration. And I don't think I'm alone in that feeling, I have never seen a Markdown editor that doesn't increment enumeration.

File: src/webrtc/vad.md (hunk 4 above)
Comment (daniel-jodlos) [reply src/webrtc/vad.md]: Depends on the target audience, I don't think it was defined anywhere. Software Engineers with at least a bachelor's degree in Computer Science? `logarithm` to your heart's content. Otherwise, I would be rather careful. We could easily end up with something incomprehensible to a lot of people.

File: src/webrtc/vad.md (hunk 1 above)
Comment (daniel-jodlos) [reply src/webrtc/vad.md]: Duplication isn't necessarily a bad thing...

File: src/webrtc/vad.md (hunk 4 above)
Comment (mickel8) [reply src/webrtc/vad.md]: Let's assume that the audience graduated from high school

File: src/webrtc/vad.md (hunk 1 above)
Comment (mickel8) [reply src/webrtc/vad.md]: Yeah but you start with:

```
All well and good, but how that translates into something readable? Well, let's start with "highest-intensity signal encodable by the payload format".
```

and that's really nice. You then explain that part and then you write 

```
Then there is an issue of decibels.
```

which suggests that now you are going to explain some remaing part of the first paragraph but in fact you are duplicating what you have written before.

In general, I feel like there are going to be two things that will be explained but in fact there is only one thing explained two times.

File: src/webrtc/vad.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
+# Active Speaker Detection
+
+A common feature in video conferencing software is active speaker detection.
+It is a small feature, but one that the user would feel weird without.
+Oftentimes, it is visualised by highlighting the user's tile while they are talking.
+Other use cases include deciding which participants should be shown on the screen.
+This can be implementedby using historical data about their voice activity.
```
Comment (mickel8) [inline src/webrtc/vad.md]:
```suggestion
This can be implemented by using historical data about their voice activity.
```

File: src/webrtc/vad.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+As you can see, voice activity detection is a small, but important piece of any WebRTC implementation.
+WebRTC standard provides tools that make it easy and convenient to implement this in your Selective Forwarding Unit.
+
+## Audio Level Header extension
+RFC 6464 defines an Audio Level Header extension, which is very useful in the context of Voice Activity Detection, as it takes the implementation load of the SFU.
```
Comment (mickel8) [inline src/webrtc/vad.md]: Can we link this RFC?

File: src/webrtc/vad.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+There are two parameters in this algorithm that you can tweak:
+1. `threshold` to tweak the sensitivity concerning the volume.
+   You should tweak it if you find that your implementation doesn't detect actual speech at all, meaning that it is too high or, on the contrary, it interprets background noise as speech.
+2. Time window duration to tweak sensitivity to detect short, but loud sounds.
+
+### Detecting silence
+Detecting silence uses the same idea of a time window and comparing average volume to the threshold as detecting speech.
+We have found it rather tricky to tweak the values in such a way that speech is detected swiftly, and silence detection isn't too aggressive.
+For that reason, we need to apply an additional condition to silence detection -a time for which the average value must be below the threshold to mark voice activity on the audio stream as silence.
```
Comment (mickel8) [inline src/webrtc/vad.md]:
```suggestion
For that reason, we need to apply an additional condition to silence detection - a time for which the average value must be below the threshold to mark voice activity on the audio stream as silence.
```

File: src/webrtc/vad.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+Please pay attention to the minus, as it makes it so that the louder it gets, the lower the value becomes.
+The fact that values are negative is a consequence of a definition of the `dBov` unit.
+
+RFC 6464 also defines an optional flag bit "V".
+When the use of it is negotiated, it indicates whether the encoder believes the audio packet contains voice activity.
+
+### What is `dBov`
+`dBov` is defined as the level (in [decibels](https://en.wikipedia.org/wiki/Decibel)) relative to the overload point of the system.
+In this context, an overload point is the highest-intensity signal encodable by the payload format.
+In simpler terms, the overload point is the loudest possible sound that can be encoded into the codec.
```
Comment (mickel8) [inline src/webrtc/vad.md]: What about putting this in a frame? Something like we have in GCC `deeper dive into RTP padding packet` and put this before information about `V` flag?

File: src/webrtc/vad.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+## Audio Level processing
+You could use the flag bit "V" if available, but we don't recommend using it for the production environment for 2 reasons:
+1. It is optional, so you cannot be sure it will always be there.
+2. Implementation isn't standardized, meaning that you can have inconsistent behavior depending on the sender.
+
+It is therefore worth considering implementing your algorithm based purely on the `level` field.
+Hopefully, you now have some understanding of the value in the `level` field, and we can jump right into the algorithm.
+Don't worry, it's not difficult at all.
```
Comment (mickel8) [inline src/webrtc/vad.md]: I wonder if this sentence wouldn't fit better at the end of `Audio Level Header extension`

**Review by mickel8 — APPROVED** [review]
